src/dodal/plans/check_topup.py

```diff
@@ -1,12 +1,28 @@
+from typing import Any
+
 import bluesky.plan_stubs as bps
 
+from dodal.common.beamlines.beamline_parameters import (
+    get_beamline_parameters,
+)
 from dodal.devices.synchrotron import Synchrotron, SynchrotronMode
 from dodal.log import LOGGER
 
 ALLOWED_MODES = [SynchrotronMode.USER, SynchrotronMode.SPECIAL]
 DECAY_MODE_COUNTDOWN = -1  # Value of the start_countdown PV when in decay mode
 COUNTDOWN_DURING_TOPUP = 0
 
+DEFAULT_THRESHOLD_EXPOSURE_S = 120
+DEFAULT_TOPUP_GATE_DELAY_S = 1
+
+
+class TopupConfig:
+    # For planned exposures less than this value, wait for topup to finish instead of
+    # collecting throughout topup.
+    THRESHOLD_EXPOSURE_S = "dodal_topup_threshold_exposure_s"
+    # Additional safety margin to wait after the end of topup.
```

Comment:
Should: A little bit more detail on why we need the safety margin would be good.